bin/asd

```diff
@@ -18,57 +20,55 @@ foreach ([__DIR__ . '/../../../autoload.php', __DIR__ . '/../vendor/autoload.php
     }
 }
 
-$options = getopt('', ['watch:', 'and-tag:', 'or-tag:', 'color:']);
-$watchMode = isset($options['watch']);
-$andTag = isset($options['and-tag']) && is_string($options['and-tag']) ? explode(',', $options['and-tag']) : [];
-$orTag = isset($options['or-tag']) && is_string($options['or-tag']) ? explode(',', $options['or-tag']) : [];
-$color = isset($options['color']) && is_string($options['color']) ? $options['color'] : null;
-$hasTag = $andTag !== [] || $orTag !== [];
-$alpsFile = realpath($argv[$argc - 1]);
+$options = getopt('c:', ['config:']);
 if ($argc === 1) {
-    echo 'usage: asd [--watch] [--and-tag=tag1,tag2] [--or-tag=tag3] [--color=red] <alps>' . PHP_EOL;
-    exit(1);
+    $options['c'] = getcwd();
 }
-if (! $alpsFile) {
-    echo 'Not readable: ' . $alpsFile . PHP_EOL;
+try {
+    $config = isset($options['c']) ? ConfigFactory::fromFile($options['c'], $argc, $argv) : ConfigFactory::fromCommandLine($argc, $argv);
+} catch (DataFileNotFoundException $e) {
+    printf('Config gile not found: %s', $e->getMessage());
+    exit(1);
+} catch (AlpsFileNotReadableException $e) {
+    printf('Profile file not found: %s', $e->getMessage());
     exit(1);
 }
-if ($watchMode) {
+if ($config->watch) {
     chdir(dirname(__DIR__) . '/asd-sync');
     $isFirstRun = ! is_dir(dirname(__DIR__) . '/asd-sync/node_modules');
     if ($isFirstRun) {
         passthru('npm install');
     }
-    passthru('npm start -- --profile ' . $alpsFile);
+    passthru('npm start -- --profile ' . $config->profile);
     exit(0);
 }
 try {
-    $profile = new Profile($alpsFile);
+    $profile = new Profile($config->profile);
     $dot = (new DrawDiagram)($profile);
     // dump docs/
-    (new DumpDocs())($profile, $alpsFile);
+    (new DumpDocs())($profile, $config->profile);
 
-    if ($hasTag) {
-        $filteredProfile = new TaggedProfile($profile, $orTag, $andTag);
-        $tagDot = isset($color) ? (new DrawDiagram())($profile, $filteredProfile, $color) : (new DrawDiagram())($filteredProfile);
+    if ($config->hasTag) {
+        $filteredProfile = new TaggedProfile($profile, $config->filter->or, $config->filter->and);
+        $tagDot = isset($config->filter->color) ? (new DrawDiagram())($profile, $filteredProfile, $config->filter->color) : (new DrawDiagram())($filteredProfile);
```

Comment:
`isset($config->filter->color)` is always true because `$config->filter->color` is type of string.(not nullable)

So, isset is not need.

```php
$tagDot = $config->filter->color ? (new DrawDiagram())($profile, $filteredProfile, $config->filter->color) : (new DrawDiagram())($filteredProfile);
Write
```